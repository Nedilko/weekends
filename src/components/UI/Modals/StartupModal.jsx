import PropTypes from 'prop-types'
import Modal from './Modal'
import DayDropdown from '../Dropdown/DayDropdown'
import HourDropdown from '../Dropdown/HourDropdown'
import { useState } from 'react'

StartupModal.propTypes = {
  onApply: PropTypes.func.isRequired,
  settings: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
}

function StartupModal({ title, settings, onApply }) {
  const [day, setDay] = useState(settings.day)
  const [hour, setHour] = useState(settings.hour)

  return (
    <Modal
      title={title}
      onApply={() => onApply({ day, hour })}
      isButtonsCentered={true}
      isHeadingCentered={true}
    >
      <div className="flex flex-col font-thin">
        <div className="flex justify-center">
          <h2 className="mb-2 text-xl">Day and Hour you finish your work</h2>
        </div>
        <div className="flex items-center justify-around">
          <DayDropdown selectedValue={day} onChange={setDay} />
          <HourDropdown selectedValue={hour} onChange={setHour} />
        </div>
      </div>
    </Modal>
  )
}

export default StartupModal
