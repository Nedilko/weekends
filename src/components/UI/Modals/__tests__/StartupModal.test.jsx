import StartupModal from '@UI/Modals/StartupModal'
import { render, screen, userEvent } from '@utils/test-utils'
import ReactDOM from 'react-dom'

describe('StartupModal', () => {
  const handleApply = vi.fn()

  const settings = {
    useSystemTheme: false,
    theme: 'light',
    greetingsText: 'sample greeting',
    day: 5,
    hour: 18,
    isFirstLoad: false,
  }

  beforeAll(() => {
    ReactDOM.createPortal = vi.fn((element) => {
      return element
    })
    window.HTMLElement.prototype.scrollIntoView = vi.fn()
  })

  afterEach(() => {
    ReactDOM.createPortal.mockClear()
    window.HTMLElement.prototype.scrollIntoView.mockClear()
  })

  afterAll(() => {
    ReactDOM.createPortal.mockRestore()
    window.HTMLElement.prototype.scrollIntoView.mockRestore()
  })

  it('should be in the document', async () => {
    render(
      <StartupModal
        settings={settings}
        title="sample title"
        onApply={handleApply}
      />
    )
    expect(screen.getByTestId('startup-modal')).toMatchSnapshot()
  })

  it('should handle Apply with new settings', async () => {
    const user = userEvent.setup()
    render(
      <StartupModal
        settings={settings}
        title="sample title"
        onApply={handleApply}
      />
    )

    await user.click(screen.getByText('Friday'))
    await user.click(screen.getByText('Saturday'))
    await user.click(screen.getByText('18:00'))
    await user.click(screen.getByText('17:00'))
    await user.click(screen.getByRole('button', { name: 'Apply' }))
    expect(handleApply).toHaveBeenCalledWith({
      day: 6,
      hour: 17,
    })
  })
})
