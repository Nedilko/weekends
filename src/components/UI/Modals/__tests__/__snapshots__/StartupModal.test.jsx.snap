// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartupModal should be in the document 1`] = `
<section
  class="flex flex-col font-thin"
  data-testid="startup-modal"
>
  <div
    class="flex justify-center"
  >
    <h2
      class="mb-2 text-xl"
    >
      Day and Hour you finish your work
    </h2>
  </div>
  <div
    class="flex items-center justify-around"
  >
    <div
      class="flex flex-row items-center"
    >
      <div
        class="mr-2 flex items-center justify-center"
      >
        Day
      </div>
      <div
        class="z-10 block"
      >
        <button
          class="flex h-6 items-center justify-between rounded-md border border-slate-100 pl-2 text-base font-thin focus:border-slate-300 focus:outline-none dark:border-gray-700 dark:focus:border-gray-500"
          type="button"
        >
          Friday
          <svg
            class="mr-1 h-4 w-4 text-gray-500 "
            fill="none"
            role="img"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 9l-7 7-7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="flex flex-row items-center"
    >
      <div
        class="mr-2 flex items-center justify-center"
      >
        Hour
      </div>
      <div
        class="z-10 block"
      >
        <button
          class="flex h-6 items-center justify-between rounded-md border border-slate-100 pl-2 text-base font-thin focus:border-slate-300 focus:outline-none dark:border-gray-700 dark:focus:border-gray-500"
          type="button"
        >
          18:00
          <svg
            class="mr-1 h-4 w-4 text-gray-500 "
            fill="none"
            role="img"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 9l-7 7-7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</section>
`;
